backend/src/main/java/backend/mulkkam/intake/service/IntakeAmountService.java

```diff
@@ -56,4 +62,15 @@ private Member getMember(Long id) {
         return memberRepository.findById(id)
                 .orElseThrow(() -> new CommonException(NotFoundErrorCode.NOT_FOUND_MEMBER));
     }
+
+    private void updateTargetAmountSnapshot(Member member) {
+        LocalDate today = LocalDate.now();
+        Optional<TargetAmountSnapshot> targetAmountSnapshot = targetAmountSnapshotRepository.findByMemberIdAndUpdatedAt(
+                member.getId(), today);
+        if (targetAmountSnapshot.isPresent()) {
+            targetAmountSnapshot.get().updateTargetAmount(member.getTargetAmount());
+            return;
+        }
+        targetAmountSnapshotRepository.save(new TargetAmountSnapshot(member, today, member.getTargetAmount()));
```

Comment:
```suggestion
        targetAmountSnapshotRepository.findByMemberIdAndUpdatedAt(member.getId(), today)
                .ifPresentOrElse((snapshot) -> {
                    snapshot.updateTargetAmount(member.getTargetAmount());
                }, () -> {
                    TargetAmountSnapshot snapshot = new TargetAmountSnapshot(member, today, member.getTargetAmount());
                    targetAmountSnapshotRepository.save(snapshot);
                });
```

이런 방법도 있긴 한데 가독성이 조금 떨어지는 것 같긴 해서 ..! 취사선택~